app/models/chat.rb

```diff
@@ -4,4 +4,18 @@ class Chat < ApplicationRecord
 
   validates :match, presence: true
   validates :match_id, uniqueness: true
+
+  def read_messages(user_id)
+    unread_messages(user_id).update_all(read: true)
+  end
+
+  def unread_messages_count(user_id)
+    unread_messages(user_id).count
+  end
+
+  private
+
+  def unread_messages(user_id)
+    messages.where(to: user_id, read: false)
```

Comment:
You can create a scopes on your `Message` model and then chain them here.

```
scope :to_user, ->(user_id) { where(to_id: user_id) }
scope :unread, -> { where(read: false) }
```

Then you can do `messages.to_user(user_id).unread`